lib/govspeak.rb

```diff
@@ -295,6 +329,19 @@ def render_image(image)
           doc.gsub!("<ul>", "<ol>")
           doc.gsub!("</ul>", "</ol>")
           doc.sub!("<ol>", '<ol class="legislative-list">')
+
+          footnotes = body.scan(/\[\^(\d)\]/)
```

Comment:
This will only catch the first 9 footnotes. I doubt we'd ever need more than that, but just in case, perhaps worth changing to:

```suggestion
          footnotes = body.scan(/\[\^(\d+)\]/)
```

You probably want to avoid the `footnotes[counter][0]` syntax later on too, by flattening the structure here, so that you could just use `footnotes[counter]`:

```suggestion
          footnotes = body.scan(/\[\^(\d+)\]/).flatten
```

Reply:
Thanks @ChrisBAshton I hadn't noticed that! I also forgot about the `flatten` method and looks much neater now.
 I've updated it now [here](https://github.com/alphagov/govspeak/pull/216/files#diff-acb6db6cd67285bd5380398c9e51ced8940ed7c4110706d038cbb243310e4055R347-R354)